test/service/test_request_validation.py

```diff
@@ -224,12 +224,11 @@ def test_bdbag_manifest(self):
             ]
             expected_fieldnames, expected_row = map(list, zip(*expectations))
             with open(os.path.join(zip_dir, zip_fname, 'data', 'sample.tsv'), 'r') as fh:
-                samples = iter(csv.reader(fh, delimiter='\t'))
-                row = next(samples)
+                rows = iter(csv.reader(fh, delimiter='\t'))
+                row = next(rows)
                 self.assertEqual(expected_fieldnames, row, 'Manifest headers are not configured correctly')
-                row = next(samples)
-                self.assertEqual(row, expected_row, 'Row in sample.tsv incorrect.')
-                remaining_rows = list(samples)
-                self.assertTrue(all(len(row) == len(expected_row) for row in remaining_rows),
+                rows = list(rows)
+                self.assertEqual(len(rows), 2, 'Found incorrect number of files.')  # self.bundle has 2 files
+                self.assertTrue(all(len(row) == len(expected_row) for row in rows),
                                 'Incorrect number of entries in a row.')
-                self.assertEqual(len(remaining_rows), 1, 'Found incorrect number of files.')  # self.bundle has 2 files
+                self.assertIn(expected_row, rows, 'Expected file was not found or had incorrect fields')
```

Comment:
We could freeze `rows`, convert it to a set, assert that the set is of size 2 and assert that `freeze(expected_row)` in that set. Here's how freeze() works:

```python
from azul.json_freeze import freeze
freeze(["a", 1, True]) in set(freeze([["a", 1, True],["b", 2, False]]))
True
freeze(["a", 2, True]) in set(freeze([["a", 1, True],["b", 2, False]]))
False
freeze([1, "a", True]) in set(freeze([["a", 1, True],["b", 2, False]]))
False
```